### ISSUE TITLE
WJEUInt32 WJE_Get with a non-existent parameter

### ISSUE THREAD

**Issue by fusionJose** [netmail-open/wjelement#84]
When I receive a json message in my application and I try to get a parameter (int value) with a certain name like this:
value = WJEUInt32(json_params, "name", WJE_GET, "");
I get 653340 as value, and it doesn't exist in such message (json_params object).

Is ther a way to check if that value named "name" is in the message or not? Before trying to get an incorrect value.

**Comment by minego** [comment]
tldr:
```
    if (!WJEGet(json_params, "name", NULL)) {
        printf("There is no \"name\"\n");
    }

```
Longer answer:

The last argument to WJEUInt32() is the value to return if it can't be found, and should be a uint32 and not a string. So, you can do something like this:

```
    x = WJEUInt32(json_params, "name", WJE_GET, 0);
    if (x != 0) {
        printf("The value is: %d\n", x);
    } else {
        printf("The value wasn't there.\n");
    }

```

This approach only works if there is a sane default value, or if there is a value like 0 that you know won't ever exist in the document. If that isn't the case then you'll have to use WJEGet() to check for the existence first.

If you are using WJEGet() and are worried about performance then you should store the result and use that when getting the value, like so:
```

    WJElement e;
    uint32 x;

    if ((e = WJEGet(json_params, "name", NULL)) && WJR_TYPE_NUMBER == e->type) {
        x = WJEUInt32(e, NULL, WJE_GET, 0);
        printf("The value is: %d\n", x);
    } else {
        printf("The value wasn't there.\n");
    }

```

This works because the call to WJEGet() returns a WJElement, which represents that portion of the document. It will return NULL if it doesn't find a match for the name, and you can easily check the type.
